### PR TITLE
Allow Flockmind to promote Flocktraces

### PR DESCRIPTION
**Pull request by FlameArrow57**
[QoL]

## About the PR 
This PR just allows Flockminds to promote Flocktraces to the Flockmind.

## Why's this needed? 
In case they have to quit the game, etc

### CODE REVIEW

**Review by TobleroneSwordfish — APPROVED** [review]
Should probably give the promoted trace a message or popup or something to tell them what happened.

**Comment by FlameArrow57** [comment]
promoteToFlockmind() already handles that if it looks good

File: code/datums/flock/flock.dm [hunk 1]
```diff
@@ -114,6 +114,25 @@ var/flock_signal_unleashed = FALSE
 				if(istype(host))
 					boutput(host, "<span class='flocksay'><b>\[SYSTEM: The flockmind has removed you from your previous corporeal shell.\]</b></span>")
 					host.release_control()
+		if("promote_trace")
+			var/message = "Are you sure?"
+
+			var/mob/living/intangible/flock/trace/trace_to_promote = locate(params["origin"])
+			if(!istype(trace_to_promote.loc, /mob/living/critter/flock/drone))
+				if (!trace_to_promote.client || trace_to_promote.afk_counter > FLOCK_AFK_COUNTER_THRESHOLD)
+					message += " This Flocktrace is afk."
```
Comment (amylizzle) [inline code/datums/flock/flock.dm:123]:
```suggestion
					message += " This Flocktrace is unresponsive."
```

File: code/datums/flock/flock.dm [hunk 2]
```diff
@@ -114,6 +114,25 @@ var/flock_signal_unleashed = FALSE
...
+			else
+				var/mob/living/critter/flock/drone/host = trace_to_promote.loc
+				if (!host.client || host.controller.afk_counter > FLOCK_AFK_COUNTER_THRESHOLD)
+					message += " This Flocktrace is afk."
```
Comment (amylizzle) [inline code/datums/flock/flock.dm:127]:
```suggestion
					message += " This Flocktrace is unresponsive."
```

File: code/datums/flock/flock.dm [hunk 3]
```diff
@@ -114,6 +114,25 @@ var/flock_signal_unleashed = FALSE
...
+
+			if (tgui_alert(usr, message, "Promote Flocktrace", list("Yes", "Cancel")) != "Cancel")
```
Comment (amylizzle) [inline code/datums/flock/flock.dm:129]:
```suggestion
			if (tgui_alert(ui.user, message, "Promote Flocktrace", list("Yes", "Cancel")) != "Cancel")
```

File: code/datums/flock/flock.dm [hunk 4]
```diff
@@ -114,6 +114,25 @@ var/flock_signal_unleashed = FALSE
...
+				var/choice = tgui_alert(usr, "Leave the Flock?", "Promote Flocktrace", list("No", "Yes", "Cancel"))
```
Comment (amylizzle) [inline code/datums/flock/flock.dm:130]:
```suggestion
				var/choice = tgui_alert(ui.user, "Leave the Flock?", "Promote Flocktrace", list("No", "Yes", "Cancel"))
```

**Review by amylizzle — CHANGES_REQUESTED** [review]

File: code/datums/flock/flock.dm [hunk 5]
```diff
@@ -113,6 +113,25 @@ var/flock_signal_unleashed = FALSE
...
+			var/mob/living/intangible/flock/trace/trace_to_promote = locate(params["origin"])
+			if(!istype(trace_to_promote.loc, /mob/living/critter/flock/drone))
+				if (!trace_to_promote.client || trace_to_promote.afk_counter > FLOCK_AFK_COUNTER_THRESHOLD)
+					message += " This Flocktrace is unresponsive."
+			else
+				var/mob/living/critter/flock/drone/host = trace_to_promote.loc
+				if (!host.client || host.controller.afk_counter > FLOCK_AFK_COUNTER_THRESHOLD)
+					message += " This Flocktrace is unresponsive."
+
+			if (tgui_alert(user, message, "Promote Flocktrace", list("Yes", "Cancel")) != "Cancel")
```
Comment (pali6) [inline code/datums/flock/flock.dm:128]: I believe you generally want to check for alert result using `==`. Closing the alert or erroring out somehow will return `null` iirc and you don't want that to act as "Yes".

File: code/datums/flock/flock.dm [hunk 6]
```diff
@@ -113,6 +113,25 @@ var/flock_signal_unleashed = FALSE
...
+				var/choice = tgui_alert(user, "Leave the Flock?", "Promote Flocktrace", list("No", "Yes", "Cancel"))
+				if (choice != "Cancel")
```
Comment (pali6) [inline code/datums/flock/flock.dm:130]: Ditto.

**Review by pali6 — CHANGES_REQUESTED** [review]
Are there any advantages to having multiple active flockminds in a flock?

**Comment by FlameArrow57** [comment]
For promoting Flocktraces, it just keeps one Flockmind always

**Review by pali6 — APPROVED** [review]

**Pull request merged**